File: home5.1/MegaData.java
import java.util.Arrays;

public class MegaData {
    private float[] smallArray = new float[1024];
    private double[] bigArray = new double[1024 * 1024];

    public MegaData() {
        reset();
    }

    public void reset() {
        Arrays.fill(smallArray, 42.0f);
        Arrays.fill(bigArray, 42.0);
    }

    public float[] getSmallArray() {
        return smallArray;
    }

    public double[] getBigArray() {
        return bigArray;
    }
}

File: class5.1/MegaData.cpp
#include "MegaData.h"

MegaData::MegaData() {
    smallArray.fill(42.0f);
    bigArray.fill(42.0);
}

void MegaData::reset() {
    smallArray.fill(42.0f);
    bigArray.fill(42.0);
}

std::array<float, 1024>& MegaData::getSmallArray() {
    return smallArray;
}

std::array<double, 1024 * 1024>& MegaData::getBigArray() {
    return bigArray;
}
